### ISSUE TITLE
Is there a way to _only_ generate c# POCO to TS interfaces?

### ISSUE THREAD

**Issue by jsancho** [RicoSuter/NSwag#2307]
I've managed to integrate the TS interfaces that are generated for all my C# models and it works great.

I've not done the full client generation, as we already have a working client-side implementation and it would require some tuning of Nswag to work with our auth, error, retry mechanisms, etc.
For now the biggest benefit is in ensuring that our payloads are in the correct shape when communicating with our API.

So ideally, I'd like to reduce the output from the TypeScript generator to only be interfaces that match the original C# classes. Which would ultimately remove thousands of lines of generated code and unnecessarily confuse in our code reviews whenever a model is changed.

I've tried with a few combinations of configurations in a little c# console app, here's my latest attempt.

```
      const string endpoint = "http://localhost:5000/swagger/v1/swagger.json";

      Console.WriteLine($"Generating typescript classes from {endpoint}");

      var document = await OpenApiDocument.FromUrlAsync(endpoint);

      var settings = new TypeScriptClientGeneratorSettings
      {
        GenerateClientClasses = false,
        GenerateClientInterfaces = false,
        GenerateOptionalParameters = true
      };

     var typeGeneratorSettings = new TypeScriptGeneratorSettings
      {
        DateTimeType = TypeScriptDateTimeType.String,
        GenerateConstructorInterface =false,
        MarkOptionalProperties = true,
        TypeStyle = TypeScriptTypeStyle.Interface
     };

      var typeResolver = new TypeScriptTypeResolver(typeGeneratorSettings);

      var generator = new TypeScriptClientGenerator(document, settings, typeResolver);
      var code = generator.GenerateFile();
      File.WriteAllText("exportedModels.ts", code);
```

I suppose that I'm also misusing some of the settings above as some don't seem to have any effect of the different outputs I've generated.

Is this a supported scenario?
Could you give us some hints on how to approach this?

Thanks,

Jaume

**Comment by RicoSuter** [comment]
You need to set change the typescript DTO settings on the client settings to interface.

Something like:

```
  var settings = new TypeScriptClientGeneratorSettings
  {
        TypeScriptSettings = typeGeneratorSettings 

```

**Comment by jsancho** [comment]
> You need to set change the typescript DTO settings on the client settings to interface.
> 
> Something like:
> 
> ```
>   var settings = new TypeScriptClientGeneratorSettings
>   {
>         TypeScriptSettings = typeGeneratorSettings 
> ```

Hi Rico, thanks for the reply.

Looking at the source in [TypeScriptClientGeneratorSettings.cs](https://github.com/RicoSuter/NSwag/blob/85ae862fd6d68173a201a79e0ad06e0be2ec5de1/src/NSwag.CodeGeneration.TypeScript/TypeScriptClientGeneratorSettings.cs), I don't see the options that you mention.

There is a `TypeScriptGeneratorSettings` property, but it's only a getter.

Then if you look the snippet in my first question, I found a different way of accessing the settings for the typescript generator, but I've not found anything option regarding DTOs that produces the output that I'm looking for.

Not sure if I'm missing something else though?

**Comment by RicoSuter** [comment]
Its this 
https://github.com/RicoSuter/NSwag/blob/85ae862fd6d68173a201a79e0ad06e0be2ec5de1/src/NSwag.CodeGeneration.TypeScript/TypeScriptClientGeneratorSettings.cs#L31

**Comment by jsancho** [comment]
thanks again for your input Rico.

I do appreciate that there is a settings class, but I did not not manage to find any combination of options to achieve the result that I wanted.

I've ended up using this nifty library https://github.com/reinforced/Reinforced.Typings for the specific use cases that we have when creating typing definitions from C# POCOs to typescript.

It has a simple fluent interface for configuration and it allows to define overrides for any specific cases that break the norm.

Another great benefit is that the **typings are generated at build-time.** 
So not requiring a running web host makes it much easier to integrate from a CI/CD perspective.

We'll keep using Nswag for the actual generation of the Swagger Web UI, it's absolutely great for that.
Nevertheless, I'd strongly recommend having a look at [RT](https://github.com/reinforced/Reinforced.Typings) if someone has specific typescript generation scenarios.
